docs/guide/arrays.md

```diff
@@ -0,0 +1,127 @@
+# Arrays
+
+Use arrays to perform an operation (or call a function) on multiple cells at a time.
+
+## About arrays
+
+In HyperFormula, an array can be:
+* A range of cell addresses (e.g. `A1:A10`)
+* An **ad-hoc array**: an input array with no ground-truth range that defines the values (e.g. `{1,3,5}`)
+* A result of an arithmetic operation (e.g. `5*A1:B5`)
+* A result of a function (e.g. `=ARRAYFORMULA(ARRAY_CONSTRAIN(A2:E5,2,2))`)
+
+An array is inherently a two-dimensional object.
+
+`1`x`1` arrays are treated as single, zero-dimensional values (**scalars**).
+
+## Enabling the array arithmetic mode
+
+To use arrays in HyperFormula, you need to enable the **array arithmetic mode**.
+
+You can enable the array arithmetic mode:
+* [Locally](#enabling-the-array-arithmetic-mode-locally) (for an individual function or operation)
+* [Globally](#enabling-the-array-arithmetic-mode-globally) (for your HyperFormula instance)
+
+### Enabling the array arithmetic mode locally
+
+To enable the array arithmetic mode once, within a particular function or formula, use the `ARRAYFORMULA` function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `ARRAYFORMULA(your_array_formula)` | `=ARRAYFORMULA(A2:A5*B2:B5)` |
+| `ARRAYFORMULA(YOUR_FUNCTION(your_array_formula))` | `=ARRAYFORMULA(ISEVEN(A2:A5*10))` |
+
+### Enabling the array arithmetic mode globally
+
+To enable the array arithmetic mode by default, everywhere in your HyperFormula instance:
+
+* In your HyperFormula [configuration](../api/interfaces/configparams.html#usearrayarithmetic), set the `useArrayArithmetic` option to `true`.
+
+With the array arithmetic mode enabled globally, you can operate on arrays without using the `ARRAYFORMULA` function:
+
+```
+=A2:A5*B2:B5
+
+ISEVEN(A2:A5*10)
+```
+
+## Array features
+
+HyperFormula lets you:
+* [Operate on arrays](#operating-on-arrays) just like on [scalars](#about-arrays)
+* [Pass arrays to functions](#passing-arrays-to-scalar-functions) that accept [scalars](#about-arrays)
+* [Broadcast](#broadcasting) a smaller input array across a larger output area
+* [Filter an array](#filtering-an-array) based on boolean arrays
+* [Constrain](#constraining-an-array-s-size) an array's size
```

Comment:
Those 2 are two very specific options that are corresponding to output of specific functions (`FILTER` and `ARRAY_CONSTRAIN`), while previous 3 are quite general features of the engine.

Reply:
@izulin - I seperated them: 5e3e2d2